### PR TITLE
[MINOR] improve(doc): Improve the index to reorganize the order of the docs

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR slightly changes the order the docs in index doc.

### Why are the changes needed?

To make the document clearer.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/how-to-install.md [hunk 1]
```diff
@@ -123,7 +123,7 @@ Gravitino publishes the Docker image to [Docker Hub](https://hub.docker.com/r/da
 You can run the Gravitino Docker image by:
 
 ```shell
-docker run -d -i -p 8090:8090 datastrato/gravitino:0.3.0-SNAPSHOT
+docker run -d -i -p 8090:8090 datastrato/gravitino:<version>
```
Comment (justinmclean) [inline docs/how-to-install.md:126]: How do you know what versions exist? Might be good to add a note on how to find them?

File: docs/index.md [hunk 2]
```diff
@@ -101,6 +101,21 @@ way. to use the Trino connector, please see:
...
   Trino connector.
 
+### Server administration
+
+Gravitino provides several ways to configure and manage the Gravitino server. Please see:
+
+* [Security](./security.md): provides the security configurations for Gravitino, including HTTPS
+  and OAuth2 configurations.
+* [Gravitino metrics](./metrics.md): provides the metrics configurations and detailed metrics list
+  of Gravitino server.
```
Comment (justinmclean) [inline docs/index.md:111]: of the

File: docs/index.md [hunk 3]
```diff
@@ -101,6 +101,21 @@ way. to use the Trino connector, please see:
...
+
+### Programming guides
+
+* [Gravitino Open API](./api/rest/gravitino-rest-api): provides the complete Open API definition of
+  Gravitino.
+* [Gravitino Javadoc](pathname:///docs/0.3.0/api/java/index.html): provides the Javadoc for Gravitino API.
```
Comment (justinmclean) [inline docs/index.md:117]: for the

File: docs/index.md [hunk 4]
```diff
@@ -61,37 +81,17 @@ to learn how to use the playground.
...
-* [Lakehouse Iceberg catalog](./lakehouse-iceberg-catalog.md): a complete guide to use Gravitino
+* [Iceberg catalog](./lakehouse-iceberg-catalog.md): a complete guide to use Gravitino to
   manage Apache Iceberg data.
-* [How to set up Gravitino Apache Iceberg REST catalog service](./iceberg-rest-service.md): a
+* [Iceberg REST catalog service](./iceberg-rest-service.md): a
   complete guide to use Gravitino as Apache Iceberg REST catalog service.
-* [Apache Hive catalog](./apache-hive-catalog.md): a complete guide to use Gravitino manage Apache Hive data.
-* [JDBC MySQL catalog](./jdbc-mysql-catalog.md): a complete guide to use Gravitino manage MySQL data.
-* [JDBC PostgreSQL catalog](./jdbc-postgresql-catalog.md): a complete guide to use Gravitino manage PostgreSQL data.
+* [Hive catalog](./apache-hive-catalog.md): a complete guide to use Gravitino manage Apache Hive data.
```
Comment (justinmclean) [inline docs/index.md:92]: ...to using Gravitino to manage Apache Hive data.

File: docs/index.md [hunk 5]
```diff
@@ -61,37 +81,17 @@ to learn how to use the playground.
...
+### Catalogs
 
 Gravitino supports different catalogs to manage the metadata in different sources. Please see:
 
-* [Lakehouse Iceberg catalog](./lakehouse-iceberg-catalog.md): a complete guide to use Gravitino
+* [Iceberg catalog](./lakehouse-iceberg-catalog.md): a complete guide to use Gravitino to
   manage Apache Iceberg data.
-* [How to set up Gravitino Apache Iceberg REST catalog service](./iceberg-rest-service.md): a
+* [Iceberg REST catalog service](./iceberg-rest-service.md): a
   complete guide to use Gravitino as Apache Iceberg REST catalog service.
```
Comment (justinmclean) [inline docs/index.md:91]: using Gravitino as an Apache Iceberge

File: docs/index.md [hunk 6]
```diff
@@ -61,37 +81,17 @@ to learn how to use the playground.
...
-* [Apache Hive catalog](./apache-hive-catalog.md): a complete guide to use Gravitino manage Apache Hive data.
-* [JDBC MySQL catalog](./jdbc-mysql-catalog.md): a complete guide to use Gravitino manage MySQL data.
-* [JDBC PostgreSQL catalog](./jdbc-postgresql-catalog.md): a complete guide to use Gravitino manage PostgreSQL data.
+* [Hive catalog](./apache-hive-catalog.md): a complete guide to use Gravitino manage Apache Hive data.
+* [MySQL catalog](./jdbc-mysql-catalog.md): a complete guide to use Gravitino manage MySQL data.
```
Comment (justinmclean) [inline docs/index.md:93]: to using Gravitino to manage

File: docs/index.md [hunk 7]
```diff
@@ -61,37 +81,17 @@ to learn how to use the playground.
...
+* [PostgreSQL catalog](./jdbc-postgresql-catalog.md): a complete guide to use Gravitino manage PostgreSQL data.
```
Comment (justinmclean) [inline docs/index.md:94]: to using Gravitino to manage

File: docs/index.md [hunk 8]
```diff
@@ -101,6 +101,21 @@ way. to use the Trino connector, please see:
 * [How to use Gravitino Trino connector](./trino-connector/index.md): a complete guide to use Gravitino
   Trino connector.
 
+### Server administration
+
+Gravitino provides several ways to configure and manage the Gravitino server. Please see:
+
+* [Security](./security.md): provides the security configurations for Gravitino, including HTTPS
```
Comment (justinmclean) [inline docs/index.md:108]: configuration

File: docs/index.md [hunk 9]
```diff
@@ -101,6 +101,21 @@ way. to use the Trino connector, please see:
...
+  and OAuth2 configurations.
+* [Gravitino metrics](./metrics.md): provides the metrics configurations and detailed metrics list
```
Comment (justinmclean) [inline docs/index.md:110]: configuration

File: docs/how-to-install.md (hunk 1 above)
Comment (jerryshao) [reply docs/how-to-install.md:126]: I describe the gravitino docker hub address above, and we will have a docker image release for each version, user could find them on the docker hub.

File: docs/index.md (hunk 8 above)
Comment (jerryshao) [reply docs/index.md:108]: There's not one security configuration. So using plural and delete "the" may be correct?

Comment (jerryshao) [reply docs/index.md:108]: There's not one security configuration. So using plural and delete "the" may be correct?

**Comment by jerryshao** [comment]
@justinmclean please help to review again, thanks.

**Review by justinmclean — COMMENTED** [review]
Your changes look good, there are some minor issues in other parts of the document that I'll fix once this is merged.

**Comment by jerryshao** [comment]
Thanks, would you please approve it @justinmclean , thanks.

File: docs/index.md (hunk 8 above)
Comment (justinmclean) [reply docs/index.md]: deleting the would be better, but it's a minor issue

**Review by justinmclean — APPROVED** [review]

**Pull request merged**